proposals/0454-task-naming-api.md

```diff
@@ -0,0 +1,231 @@
+# Task Naming API
+
+* Proposal: [SE-NNNN](NNNN-filename.md)
+* Authors: [Konrad Malawski](https://github.com/ktoso), [Harjas Monga](https://github.com/Harjas12)
+* Review Manager: TBD
+* Status: **Awaiting implementation** or **Awaiting review**
+* Implementation: Work In Progress [PR](https://github.com/swiftlang/swift/pull/77609)
+* Review: ([pitch](https://forums.swift.org/...))
+
+## Introduction
+
+In this proposal, we introduce several new APIs to allow developers to name their Swift Tasks for the purposes of identifying tasks in a human readable way manually and by various developer tools.
+
+## Motivation
+
+In previous generations of concurrency technologies, developer tools, such as debuggers, have had access to some kind of label to help describe a process’s concurrent work. Ex: Pthread names or GCD queue names. These names are very helpful to provide extra context to developers when using debugging and profiling tools. 
+
+Currently, Swift Concurrency has no affordances to allow developers to label a Task, which can be troublesome for developers trying to identify "which task" is taking a long time to process or similar questions when observing the system externally. In order to ease the debugging and profiling of Swift concurrency code, developers should be able to annotate their Swift Tasks to describe an asynchronous workload.
+
+## Proposed solution
+
+In order to allow developers to provide helpful names for Swift Tasks, the Swift Task creation APIs should be modified to *optionally* allow developers to provide a name for that task.
+
+Consider the example:
+```swift
+let getUsers = Task {
+	await users.get(accountID))
+}
+```
+
+In order to ease debugging, a developer could create this unstructured task by passing in a name instead:
+
+```
+let getUsers = Task(name: "Get Users") {
+	await users.get(accountID)
+}
+```
+
+Or, if a developer has a lot of similar Tasks, they can provide more contextual information using string interpolation.
+
+```
+let getUsers = Task("Get Users for \(accountID)") {
+	await users.get(accountID)
+}
+```
+
+By introducing this in a structured, coherent way in Swift itself, rather than developers each inventing their own task-local with a name, runtime inspection tools and debuggers can become aware of task names and show you exactly which accountID was causing the crash or a profiling tool could tell you which accountID request was slow to load.
+
+## Detailed design
+
+
+Naming tasks is only allowed during their creation, and modifying names is not allowed. Names are arbitrary user-defined strings, which may be computed at runtime because they often contain identifying information such as the request ID or similar runtime information.
+
+In order to allow naming tasks, the following APIs will be provided on `Task`:
+
+```swift
+  init(
+     name: String? = nil,
```

Comment:
```suggestion
     name: String?,
```

Reply:
If I may ask: is something you both would consider for the naming API to use an enum? I am asking this, because it would be also good to in case of having to iterate over the Tasks we may have, comparing them or even making using a `switch`. Of course, my suggestion is not to remove the String as a type for naming it, but having a second flexible way to deal with them. 

What do you think?